### PR TITLE
Feature parse 10x dir

### PR DESCRIPTION
**Pull request by a-solovyev12**
- parsing 10X directories directly rather than separate matrix, barcode and genes files
- allow for different training and prediction datasets (initially single dataset was assumed)
- adapting scripts to common input from meta-workflow

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,17 @@ Graphical representation of the general workflow:
 
 ![](https://github.com/ebi-gene-expression-group/garnett-cli/blob/master/garnett_pipeline.png)
 
+## Installation 
+Garnett-cli is installed through conda: 
+```
+conda create --name <garnett-env>
```
Comment (pinin4fjords) [inline README.md]: Probably mention that you need to have Bioconda installed properly with the correct channels etc. Something I realised the other day testing something else with Anja. See https://github.com/ebi-gene-expression-group/experiment_metadata.

File: garnett_check_markers.R [hunk 2]
```diff
@@ -1,8 +1,6 @@
 #!/usr/bin/env Rscript
 
-# Load optparse we need to check inputs
```
Comment (pinin4fjords) [inline garnett_check_markers.R]: Can we keep the comments please?

File: garnett_check_markers.R [hunk 3]
```diff
@@ -145,11 +141,11 @@ marker_check = check_markers(cds = cds,
                              classifier_gene_id_type = opt$classifier_gene_id_type)
 
 if(! is.na(opt$plot_output_path)){
-    print(paste("plotting path", opt$plot_output_path))
+    print(paste("plotting path ", opt$plot_output_path))
```
Comment (pinin4fjords) [inline garnett_check_markers.R]: You don't to add a space- paste() will provide one

File: garnett_classify_cells.R [hunk 4]
```diff
@@ -1,8 +1,6 @@
 #!/usr/bin/env Rscript
 
-# Load optparse we need to check inputs
```
Comment (pinin4fjords) [inline garnett_classify_cells.R]: Again, please retain comments

File: parse_expr_data.R [hunk 5]
```diff
@@ -1,77 +1,69 @@
 #!/usr/bin/env Rscript
 
-# Load optparse we need to check inputs
 suppressPackageStartupMessages(require(optparse))
-# Load common functions
 suppressPackageStartupMessages(require(workflowscriptscommon))
+suppressPackageStartupMessages(require(garnett))
```
Comment (pinin4fjords) [inline parse_expr_data.R]: Please move the package loads (except for optparse) back to after argument parsing and checking. No point spending time doing the load until we've done those checks.

File: parse_expr_data.R [hunk 6]
```diff
@@ -1,77 +1,69 @@
...
-                                                'phenotype_data',
-                                                'feature_data', 
-                                                'output_file'))
+opt = wsc_parse_args(option_list)
+input_dirs = c(opt$ref_10x_dir, opt$query_10x_dir)
+cds_names = c(opt$ref_output_cds, opt$query_output_cds)
+# process input directories
+for(idx in 1:length(input_dirs)){
+    input_dir = input_dirs[idx] 
+    if(! file.exists(input_dir)) stop(paste('File ', input_dir, 'does not exist'))
```
Comment (pinin4fjords) [inline parse_expr_data.R]: Please separate checking of inputs from the rest of the logic. We should check everything is ready to go before proceeding with anything.

File: parse_expr_data.R [hunk 7]
```diff
@@ -1,77 +1,69 @@
...
-
-if(! file.exists(opt$phenotype_data)){
-    stop((paste('File ', opt$phenotype_data, 'does not exist')))
-}
+    # parse individual files into CDS object 
+    expr_matrix = Matrix::readMM(paste(input_dir, "/matrix.mtx", sep=""))
+    genes = read.table(paste(input_dir, "/genes.tsv", sep=""), sep="\t")
+    barcodes = read.table(paste(input_dir, "/barcodes.tsv", sep=""), sep="\t")
+    # matrix entries need to be named
+    #row.names(expr_matrix) = genes[, 1]
```
Comment (pinin4fjords) [inline parse_expr_data.R]: Remove comment lines :-)

**Comment by pinin4fjords** [comment]
Good work @a-solovyev12. As noted in comments above, I'd just like us to retain the pattern of:

1. Load optparse
2. Check inputs are valid and any input files/ dirs exist
3. Load rest of packages required
4. Do rest of logic

Thanks!

**Comment by a-solovyev12** [comment]
@pinin4fjords Thanks for your feedback!

**Pull request merged**